app/src/main/java/org/stepik/android/view/lesson/ui/activity/LessonActivity.kt

```diff
@@ -311,6 +371,17 @@ class LessonActivity : FragmentActivityBase(), LessonView, NextMoveable, RateApp
         screenManager.openTextFeedBack(this, supportEmailData)
     }
 
+    override fun onTimeIntervalPicked(data: Intent) {
+        val intervalCode = data.getIntExtra(TimeIntervalPickerDialogFragment.RESULT_INTERVAL_CODE_KEY, TimeIntervalUtil.defaultTimeCode)
+        lessonPresenter.setStreakTime(intervalCode)
+        analytic.reportEvent(Analytic.Streak.CHOOSE_INTERVAL, intervalCode.toString())
+        Snackbar.make(lessonPager, R.string.streak_notification_enabled_successfully, Snackbar.LENGTH_LONG)
+                .setTextColor(ColorUtil.getColorArgb(R.color.white, baseContext))
+                .show()
+
+        // TODO StepAttemptFragment handled a cancelled state by showing a snackbar saying that the notifications were not enabled
```

Comment:
We can add another method to `TimeIntervalPickerDialogFragment.Callback` to handle it

Reply:
Also it strange that `TimeIntervalPickerDialogFragment` uses Callback and onActivityResult logic at the same time

Reply:
It seems that current callback logic is broken and doesn't handle configuration changes. It should be fixed or removed and only activity result used.

Reply:
I have decided to removed the onActivityResult logic. The callback logic now handles configuration changes, so it should work fine